src/components/shared/TokenAMMPriceBadge.tsx

```diff
@@ -0,0 +1,88 @@
+import { LoadingOutlined } from '@ant-design/icons'
+import { useContext } from 'react'
+import { ThemeContext } from 'contexts/themeContext'
+import { useUniswapPriceQuery } from 'hooks/ERC20UniswapPrice'
+import { formattedNum } from 'utils/formatNumber'
+import UniswapLogo from 'components/icons/Uniswap'
+import { Tooltip } from 'antd'
+import { t } from '@lingui/macro'
+
+type exchangeName = 'Uniswap'
+
+const LOGOS = {
+  Uniswap: UniswapLogo,
+}
+
+const QUERIES = {
+  Uniswap: useUniswapPriceQuery,
+}
+
+type Props = {
+  exchangeName: exchangeName
+  tokenSymbol: string
+  tokenAddress: string
+  exchangeLink: string
+}
+
+export default function TokenAMMPriceBadge({
+  exchangeName,
+  tokenSymbol,
+  tokenAddress,
+  exchangeLink,
+}: Props) {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+  const Logo = LOGOS[exchangeName]
```

Comment:
other nit: the convention we've been using is lowercase for variable names and Titlecase for Types/Components, so let's make `Logo` > `logo` and `LOGOS` -> `logos`?

love this pattern for making this component compatible with multiple exchanges! can't wait to see it with Sushi etc

Reply:
Might be personal pref. but I had it capitalized because it's a component. I renamed it to `LogoComponent` - perhaps that's more clear. Don't have a strong opinion though - happy to change to `logo` 👍

Reply:
ahhh, true, could go either way here. okay, i'll take the L.

Reply:
>  i'll take the L
LoL